app/services/from_address_creation.rb

```diff
@@ -68,4 +69,8 @@ def use_default_email?
     from_address.email.blank? ||
       from_address.email == FromAddress::DEFAULT_EMAIL_FROM
   end
+
+  def allowed_email?
```

Comment:
`delegate :allowed_domain? to: :from_address`

as mentioned above. then this method can come out

Reply:
all done